Neos.Flow/Classes/ResourceManagement/Target/FileSystemTarget.php

```diff
@@ -290,11 +314,19 @@ protected function publishFile($sourceStream, $relativeTargetPathAndFilename)
         }
 
         $targetPathAndFilename = $this->path . $relativeTargetPathAndFilename;
+        $streamMetaData = stream_get_meta_data($sourceStream);
+        $sourcePathAndFilename = $streamMetaData['uri'];
```

Comment:
IMHO this is error prone as we cannot assume a stream with URI in all cases. So we need to at least null coalesce here.

```suggestion
        $sourcePathAndFilename = $streamMetaData['uri'] ?? null;
```